omod-1.11/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_11/LivingPatientSearchHandler1_11.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.webservices.rest.web.v1_0.search.openmrs1_11;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LivingPatientSearchHandler1_11 implements SearchHandler {
+	
+	private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1 + "/patient",
+	        Arrays.asList("1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*"),
+	        Arrays.asList(new SearchQuery.Builder("Allows you to find all patients including the dead or only living patients")
+	                .withRequiredParameters("q").withOptionalParameters("includeDead").build()));
+	
+	@Override
+	public SearchConfig getSearchConfig() {
+		return searchConfig;
+	}
+	
+	@Override
+	public PageableResult search(RequestContext context) throws ResponseException {
+		String includeDeadStr = context.getParameter("includeDead");
+		String searchPhrase = context.getParameter("q");
+		Boolean includeDead = StringUtils.isNotBlank(includeDeadStr) ? Boolean.parseBoolean(includeDeadStr) : false;
+		if (StringUtils.isNotBlank(searchPhrase)) {
+			List<Patient> allPatients = Context.getPatientService().getPatients(searchPhrase);
+			if (allPatients == null && allPatients.size() == 0)
```

Comment:
Can the above condition ever be true?

Reply:
Can you review your pull requests line by line to catch such obvious issues? Then ask us to review.

Reply:
ok sir . i have cleared it